From 1d57f6fe7339a597d92a0fe0c99f5bce8db13814 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Marchesin?= <marcheu@chromium.org>
Date: Fri, 24 Apr 2015 15:22:02 -0700
Subject: [PATCH 02/20] CHROMIUM: drivers/iommu: Don't reserve the domains for
 display
X-NVConfidentiality: public

Tegra drm attaches the domains by itself, don't try to initialize and
attach them from the iommu driver.

Change-Id: I4e7a11fb4a960f55491817431682ec200102b3c5
---
 drivers/iommu/tegra-smmu.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/iommu/tegra-smmu.c b/drivers/iommu/tegra-smmu.c
index 6e579815bc94..f41f7b69360e 100644
--- a/drivers/iommu/tegra-smmu.c
+++ b/drivers/iommu/tegra-smmu.c
@@ -1059,6 +1059,7 @@ enum {
 	SYSTEM_DEFAULT,
 	SYSTEM_PROTECTED,
 	SYSTEM_AVPC,
+	SYSTEM_DRM,
 	NUM_OF_STATIC_MAPS,
 };
 
@@ -1077,6 +1078,10 @@ int tegra_smmu_get_asid(struct device *dev)
 		return SYSTEM_PROTECTED;
 	else if (test_bit(TEGRA_SWGROUP_AVPC, swgroups))
 		return SYSTEM_AVPC;
+	else if (test_bit(TEGRA_SWGROUP_DC, swgroups))
+		return SYSTEM_DRM;
+	else if (test_bit(TEGRA_SWGROUP_DCB, swgroups))
+		return SYSTEM_DRM;
 
 	return SYSTEM_DEFAULT;
 }
@@ -1105,6 +1110,10 @@ static int smmu_iommu_bound_driver(struct device *dev)
 		map = smmu_handle->map[SYSTEM_PROTECTED];
 	else if (test_bit(TEGRA_SWGROUP_AVPC, swgroups))
 		map = smmu_handle->map[SYSTEM_AVPC];
+	else if (test_bit(TEGRA_SWGROUP_DC, swgroups))
+		return 0;
+	else if (test_bit(TEGRA_SWGROUP_DCB, swgroups))
+		return 0;
 	else
 		map = smmu_handle->map[SYSTEM_DEFAULT];
 
@@ -1488,6 +1497,8 @@ static void tegra_smmu_create_default_map(struct smmu_device *smmu)
 	for (i = 0; i < smmu->num_as; i++) {
 		dma_addr_t base = smmu->iovmm_base;
 		size_t size = smmu->page_count << PAGE_SHIFT;
+		if (i == SYSTEM_DRM)
+			continue;
 
 		smmu->map[i] = arm_iommu_create_mapping(&platform_bus_type,
 							base, size, 0);
-- 
2.1.4

